src/runtime/storageNG/reference-mode-store.ts

```diff
@@ -441,8 +441,10 @@ export class ReferenceModeStore<Entity extends SerializedEntity, S extends Dicti
   }
 
   private addFieldToValueList(list: Dictionary<{}>, value: {}, version: Dictionary<number>) {
-    // NOTE: This could potentially be an extension point to provide automatic IDs if we
-    // need some sort of field-level collection capabilities ahead of entity mutation.
+    // construct id for primitive fields
+    if (value['id'] == undefined) {
```

Comment:
I think you're supposed to use === with undefined. I'm not sure what the required behaviour is for nulls here, but `=== undefined` seems safe and clear, and does not include nulls.